### PR TITLE
Updates references to v0.5.0 jar

### PR DESCRIPTION
**Pull request by jainr**
## Description
This PR updates the hardcoded reference in comments and examples to specific version, removed it to point to latest jar. 
Kept the reference to a specific version only for test config so right jar could be tested.

Resolves #XXX

## How was this PR tested?
No test required.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

**Comment by xiaoyongzhu** [comment]
@jainr shall we also update this to the latest version?

`FEATHR_MAVEN_ARTIFACT="com.linkedin.feathr:feathr_2.12:0.4.0"`

https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/constants.py#L31

**Comment by blrchen** [comment]
Please update release docs for cover these as well

**Comment by jainr** [comment]
> @jainr shall we also update this to the latest version?
> 
> `FEATHR_MAVEN_ARTIFACT="com.linkedin.feathr:feathr_2.12:0.4.0"`
> 
> https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/constants.py#L31

Good catch, updated it.

**Comment by jainr** [comment]
> > release
> 
> i am ok to keep an internal doc to secure credentials

Docs updated.

File: docs/dev_guide/feathr_overall_release_guide.md [hunk 1]
```diff
@@ -22,6 +28,28 @@ Read through the [commit log](https://github.com/linkedin/feathr/commits/main) t
 - improvements and changes of this release
 - new contributors of this release
 
+## Code Changes
+Before the release is made, the version needs to be updated in following places
+- [build.sbt](https://github.com/linkedin/feathr/blob/73656fe4a57219e99ff6fede10d51a000ae90fa1/build.sbt#L3) - For Maven release version
+- [setup.py](https://github.com/linkedin/feathr/blob/73656fe4a57219e99ff6fede10d51a000ae90fa1/feathr_project/setup.py#L10) - For PyPi release version
+- [conf.py](https://github.com/linkedin/feathr/blob/73656fe4a57219e99ff6fede10d51a000ae90fa1/feathr_project/docs/conf.py#L27) - For documentation version
+- [feathr_config.yaml](https://github.com/linkedin/feathr/blob/73656fe4a57219e99ff6fede10d51a000ae90fa1/feathr_project/test/test_user_workspace/feathr_config.yaml#L84) - To set the spark runtime location for Azure Synapse and Azure Databricks used by test suite.
+- [constants.py](https://github.com/linkedin/feathr/blob/73656fe4a57219e99ff6fede10d51a000ae90fa1/feathr_project/feathr/constants.py#L31) - To set the default maven artifact version
```
[the remark this answers is not in the document]
Comment (xiaoyongzhu) [reply docs/dev_guide/feathr_overall_release_guide.md:37]: like ../build.sbt

[the remark this answers is not in the document]
Comment (jainr) [reply docs/dev_guide/feathr_overall_release_guide.md:37]: I did it to have permalink to specific location in the file.

[the remark this answers is not in the document]
Comment (jainr) [reply docs/dev_guide/feathr_overall_release_guide.md:37]: Which is not possible through markdown.

**Review by xiaoyongzhu — APPROVED** [review]

File: docs/dev_guide/feathr_overall_release_guide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiaoyongzhu) [reply docs/dev_guide/feathr_overall_release_guide.md:37]: I see, make sense

**Review by blrchen — APPROVED** [review]

**Pull request merged**